app/helpers/address_helper.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module AddressHelper
+  def postcode_lookup(postcode)
+    postcode_url = "#{ENV['API_URL']}?postcode=#{postcode}&key=#{ENV['API_KEY']}"
+
+    response = Typhoeus.get(postcode_url)
+
+    results = []
+    if response.code == 200
+      addresses = JSON.parse(response.body)["results"]
+      addresses.each do |address|
+        results << { address.dig("DPA", "UPRN") => address.dig("DPA", "ADDRESS") }
```

Comment:
If you make my proposed change to add `dataset=LPI` to the query string, then `DPA` here will need changing to `LPI`.

Reply:
Thanks @brucebolt - I've update as per your suggestion.